### ISSUE TITLE
Problems with dirs and files being numbers

### ISSUE THREAD

**Issue by ubmarco** [datakortet/yamldirs#5]
Hello,

I found 2 issues when using a yaml like this:
```
    - Config:
        - 4
    - 2.15:
        - test
```
The file named "4" does not get created. The library silently fails here.
The directory named 2.15 throws this exception:
```
filemaker.py:29: in _make_item
    return self.make_dict_item(item)
filemaker.py:47: in make_dict_item
    self.mkdir(k)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <yamldirs.filemaker.Filemaker object at 0x7f0db1d18160>, dirname = 2.15

    def mkdir(self, dirname):
>       os.mkdir(dirname)
E       TypeError: mkdir: illegal type for path parameter

filemaker.py:75: TypeError
```
The following code changes made it work for me:
```
import numbers
...
    def _make_item(self, item):
        if isinstance(item, dict):
            return self.make_dict_item(item)
->      elif isinstance(item, basestring) or isinstance(item, numbers.Number):
->          return self.make_string_item(str(item))
        elif isinstance(item, list):
            return self.make_list_item(item)
...
    def make_dict_item(self, dct):
        for k, v in dct.items():
->          k = str(k)
->          if isinstance(v, basestring) or isinstance(v, numbers.Number):
                self.make_file(filename=k, content=v)
            else:
                self.mkdir(k)
                self.pushd(k)
                self._make_item(v)
                self.popd()
```

**Comment by thebjorn** [comment]
This is fixed in b85c082bbdd2f67e83c75bd1cc5f8c3003cadbe9 and released as 1.1.2 (available from PyPI).

It was possible to create file names containing digits in the previous version also by enclosing them in quotes:```

    dir1:
        - "4"
```
but I agree that that seems awkward (and is not documented).

Thanks for the code, I used your idea and extended it to other types, like dates.

**Comment by ubmarco** [comment]
Thanks again for the quick fixes.
Unfortunately, I'm getting the following exception in the `value2str` function:
```
py35_venv_common/lib/python3.5/site-packages/yamldirs/filemaker.py:52: in _make_item
    return self.make_dict_item(item)
py35_venv_common/lib/python3.5/site-packages/yamldirs/filemaker.py:73: in make_dict_item
    self.make_file(filename=k, content=self.value2str(v))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <yamldirs.filemaker.Filemaker object at 0x7f371b3c0f28>, val = None

    def value2str(self, val):
        if isinstance(val, basestring):
            return val
        elif isinstance(val, numbers.Number):
            return str(val)
        elif type(val) == datetime.date:  # datetimes are subclasses of dates.
            return val.isoformat()
        else:
            raise UnknownType("Don't know what to do with %r of type %s" % (
>               val, type(val)
            ))
E           yamldirs.filemaker.UnknownType: Don't know what to do with None of type <class 'NoneType'>

py35_venv_common/lib/python3.5/site-packages/yamldirs/filemaker.py:47: UnknownType
```
Also see attachments for some more debugging information.

![screenshot_2017-10-25_13-30-50](https://user-images.githubusercontent.com/839934/31996546-39c07fc8-b989-11e7-9b48-0b3375139085.png)
![screenshot_2017-10-25_13-32-15](https://user-images.githubusercontent.com/839934/31996550-3d3f5e30-b989-11e7-9e63-7cb6e69adcea.png)

**Comment by thebjorn** [comment]
From the README:

> an empty directory must use YAML's inline list syntax:
> 
> foo: []

:-)

**Comment by ubmarco** [comment]
Ah ok, the documentation is clear about that.
However, the syntax seems to be valid yaml as pyyaml does not complain. The problem is that the error message is not pinpointing the problem. I'm having a faily huge yaml file of several thousand lines an it's a bit tedious to find all the empty directories.

**Comment by thebjorn** [comment]
Fixed in v1.1.3 :-)

**Comment by ubmarco** [comment]
Buddy, thanks a lot. Looks like there are no more issues for me :)
